ledger/participant-integration-api/src/main/scala/platform/DispatcherState.scala

```diff
@@ -24,53 +24,54 @@ class DispatcherState(dispatcherShutdownTimeout: Duration)(implicit
 ) {
   private val logger = ContextualizedLogger.get(getClass)
   @volatile private var dispatcherStateRef: DispatcherState.State =
-    DispatcherNotInitialized
+    DispatcherNotRunning
 
-  def initialized: Boolean = dispatcherStateRef match {
-    case _: DispatcherState.DispatcherOperational => true
-    case DispatcherState.DispatcherNotInitialized | DispatcherState.DispatcherShutdown => false
+  def getDispatcher: Dispatcher[Offset] = dispatcherStateRef match {
+    case DispatcherNotRunning => throw dispatcherNotRunning()
+    case DispatcherStateShutdown => throw dispatcherStateShutdown()
+    case DispatcherRunning(dispatcher) => dispatcher
   }
 
-  def getDispatcher: Dispatcher[Offset] = dispatcherStateRef match {
-    case DispatcherState.DispatcherNotInitialized => throw dispatcherNotInitializedException()
-    case DispatcherState.DispatcherOperational(dispatcher) => dispatcher
-    case DispatcherState.DispatcherShutdown => throw dispatcherShutdownException()
+  def startDispatcher(ledgerEnd: LedgerEnd): Unit = synchronized {
+    dispatcherStateRef match {
+      case DispatcherNotRunning =>
+        val activeDispatcher = buildDispatcher(ledgerEnd)
+        dispatcherStateRef = DispatcherRunning(activeDispatcher)
+      case DispatcherStateShutdown => throw dispatcherStateShutdown()
+      case DispatcherRunning(_) =>
+        throw new IllegalStateException(
+          "Dispatcher startup triggered while an existing dispatcher is still active."
+        )
+    }
   }
 
-  def reset(ledgerEnd: LedgerEnd): Future[Unit] = synchronized {
+  def stopDispatcher(): Future[Unit] = synchronized {
     dispatcherStateRef match {
-      case DispatcherState.DispatcherNotInitialized =>
-        dispatcherStateRef = DispatcherOperational(buildDispatcher(ledgerEnd))
-        Future.unit
-      case DispatcherOperational(dispatcher) =>
-        dispatcherStateRef = DispatcherOperational(buildDispatcher(ledgerEnd))
-        shutdownDispatcher(dispatcher)
-      case DispatcherState.DispatcherShutdown =>
-        Future.failed(dispatcherShutdownException())
+      case DispatcherNotRunning | DispatcherStateShutdown => Future.unit
+      case DispatcherRunning(dispatcher) =>
+        dispatcherStateRef = DispatcherNotRunning
+        logger.info("Stopping active Ledger API offset dispatcher.")
+        dispatcher
+          // TODO LLP: Fail sources with exception instead of graceful shutdown
```

Comment:
I think it would be important to do this, because AFAIK this can trigger further emitting until the last ledger-end.
Also shutdown needs really this, stopping probably not

Reply:
Done.